File: scriptedmain.lua
#!/usr/bin/env lua

function main(arg)
	print("Program: " .. debug.getinfo(1).source)

	print("Number of Args: " .. #arg)

	for i,a in ipairs(arg) do
		print("Arg: " .. a)
	end

end

if type(package.loaded[(...)]) ~= "userdata" then
	main(arg)
else
	module(..., package.seeall)
end

File: scriptedmain.R
#!/usr/bin/Rscript

getProgram <- function(args) {
	sub("--file=", "", args[grep("--file=", args)])
}

main <- function(program, args) {
	cat("Directory: ", getwd(), "\n")

	cat("Program: ", program, "\n")

	cat("Number of Args: ", length(args), "\n")

	for (arg in args) {
		cat("Arg: ", arg, "\n")
	}
}

args <- commandArgs(trailingOnly = FALSE)
program <- getProgram(args)

if (length(program) > 0 && length(grep("scriptedmain", program)) > 0) {
	main(program, args)
}

q("no")